background.js

```diff
@@ -1,33 +1,75 @@
-const showMessage = async (message) => {
+const setLoadingState = async (active) => {
+  const activeCss = `.caal-loading {
+    position: absolute;
+    display: flex;
+    background: #85a94e;
+    padding: 1em;
+
+    transform: translateX(-50%);
+    left: 50%;
+    margin-top: 40vh;
+
+    z-index: 2000;
+
+    box-shadow: 0 var(--borderHeight) 0 0 var(--borderColor),0 .7rem .95rem .05rem var(--secondaryBorderColor);
+    border-radius: var(--borderRadius,.5rem);
+
+    color: #fff;
+    font-family: var(--globalSecondaryFont);
+    font-size: 3rem;
+    line-height: 1;
+    font-weight: 700;
+    text-shadow: 0 .1rem 0 rgba(0,0,0,.4);
+  }`;
+
+  const deactiveCss = `.caal-loading {
+    display: none;
+  }`;
+
+  browser.tabs.insertCSS({ code: active ? activeCss : deactiveCss });
```

Comment:
Wouldn't it make more sense to:

1. Always include styles for both classes in the DOM
2. Conditionally set a class that will disable or enable the component

This way you only change an element's class rather than keep injecting new CSS stylesheets. I have not seen anyone dynamically change styles by inserting new stylesheets :smile: Changing the classes seems like the way to go, unless I am missing something

If it is hard to do (and I can see that it can be), that may be a sign that the extension is not using the right tools for the job. To modify the page, it could be using a content script that could control the element's class. It is probably harder to control everything from a single background script.

Reply:
Component was moved to content script. Now implementation is more elegant. Thanks for informing about this possibility :smile: